### DISCUSSION TITLE
Recommendations for catching exceptions

### DISCUSSION THREAD

**Discussion by njk42** [ArthurSonzogni/FTXUI#451] [General]
I'm really liking the terminal-based UI capabilities which this library provides, and am starting to get a decent feel for how all of the components, elements, containers, renderers, etc. come together hierarchically and functionally.  Thank you for the effort you've put into developing it!

However, I'm not finding any information in the examples or documentation recommending how to handle exceptions which might be thrown from within one of the functions (often a lambda) that one attaches to a UI element like a Button.on_click() or CheckboxOption.on_change().  (e.g. Where would one put a Try/Catch block?)

It would be great to see this prototyped into a Modal element to show an error message when an exception is caught, or alternatively provide some other "Best-Practice" for handling the case where program logic gets invoked from within a GUI element but doesn't function as desired and an exception or error information needs to be relayed to the user.  Thanks!

**Comment by ArthurSonzogni** [comment]
Thanks @njk42 for using FTXUI and this message!

I am following the Chromium/[Google C++ Style guide](https://google.github.io/styleguide/cppguide.html#Exceptions). As such, I do not use C++ exceptions.

FTXUI library and examples compile without compiler support for exceptions.
I don't have a good recommendation for you, other than not using exception. Is exception the only way you have to deal with your code logic?

**Reply by njk42** [reply]
Thanks for your reply @ArthurSonzogni.  Certainly exceptions can be handled (for any calling functions which throw) directly within try/catch blocks local to the invoked lambda associated with an .on_click or .on_change.  The alternative would typically be converting an exception to a return code indicating an error, but this is also not convenient or possible to do within the framework.  So that's why I was hoping someone else had a better design pattern or example to recommend for how best to propagate error information to users on the UI from within an arbitrary place inside the screen.loop(component) thread, whether via a build-in behavior like CatchEvent or manually implemented via global flags/strings.
